src/map.cpp

```diff
@@ -139,16 +139,29 @@ Map::Map(const SizedType &type) : IMap(type)
   }
 }
 
-Map::Map(libbpf::bpf_map_type map_type) : IMap(map_type)
+Map::Map(libbpf::bpf_map_type map_type, uint64_t perf_rb_pages) : IMap(map_type)
```

Comment:
This whole thing of using different `Map` constructors for different kinds of maps is terribly ugly, but you could probably follow it and create a separate constructor `Map(libbpf::bpf_map_type, uint64_t)` for ringbuf maps 
while using the original constructor `Map(libbpf::bpf_map_type)` for perf events map.

Anyways, this should be refactored as it took me a while to figure out what's going on.